src/core/error_helpers.js

```diff
@@ -105,6 +110,159 @@ p5._friendlyFileLoadError = function (errorType, filePath) {
+    for (var i = 0; i < arrDoc.length; i++) {
+      errorListArray.push(testParamFormat(args, arrDoc[i]));
+    }
+    // compare errors from all formats
+    var minErrInd = -1;
+    var minErrCount = 0;
+    for (var j = 0; j < errorListArray.length; j++) {
+      var numErr = errorListArray[j].length;
+      if (numErr >= arrDoc.length){   // exclude clean cases
+        if(j===0){  // the first case
+          minErrCount = numErr;
+          minErrInd = j;
+        }else{
+          if(minErrCount > numErr){
+            minErrInd = j;
+          }
+        }
+      }
+    }
+    if (minErrInd > -1){
+      // generate error msg with a smaller number of errors
+      for(var n = 0; n < errorListArray[minErrInd].length; n++) {
+        p5._friendlyParamError(errorListArray[minErrInd][n], func);
+      }
+    }
+  } else {
+    var errorArray = testParamFormat(args, arrDoc[0]);
+    if (errorArray.length > 0){
+      for(var m = 0; m < errorArray.length; m++) {
+        p5._friendlyParamError(errorArray[m], func);
+      }
+    }
+  }
+}
+// validateParameters() helper functions:
+// lookupParamDoc() for querying data.json
+function lookupParamDoc(func){
+  var queryResult = arrDoc.classitems.
+    filter(function (x) { return x.name === func; });
+  if (queryResult[0].hasOwnProperty('overloads')){
+    var res = [];
+    for(var i = 0; i < queryResult[0].overloads.length; i++) {
+      res.push(queryResult[0].overloads[i].params);
+    }
+    return res;
+  } else {
+    return [queryResult[0].params];
+  }
+}
+function testParamFormat(args, format){
+  var errorArray = [];
+  var error;
+  for (var p = 0; p < format.length; p++) {
+    var argType = typeof(args[p]);
+    if ('undefined' === argType || null === argType) {
+      if (format[p].optional !== true) {
+        error = {type:'EMPTY_VAR', position: p};
+        errorArray.push(error);
+      }
+    } else {
+      var types = format[p].type.split('|'); // case accepting multi-types
+      var pass;
+      if (argType === 'object'){             // if obejct, test for class
+        pass = testParamClass(args[p], types);
+        if (!pass) {  // if fails to pass
+          error = {type:'WRONG_CLASS', position: p,
+            correctClass: types[p], wrongClass: args[p].name};
+          errorArray.push(error);
+        }
+      }else{                                 // not object, test for type
+        pass = testParamType(args[p], types);
+        if (!pass) {  // if fails to pass
+          error = {type:'WRONG_TYPE', position: p,
+            correctType: types[p], wrongType: argType};
+          errorArray.push(error);
+        }
+      }
+    }
+  }
+  return errorArray;
+}
+// testClass() for object type parameter validation
+// Returns true if PASS, false if FAIL
+function testParamClass(param, types){
+  var result = false;
+  for (var i = 0; i < types.length; i++) {
+    if (param.name === types[i]) {
+      result = true;    // class name match, pass
```

Comment:
we could change these lines to `return true`, then outside of the for loop have a `return false` line. this would add two small performance improvements (1) no need to create variable called result each time, and (2) we can break out of the loop as soon as a true value is found.
this is the same for testParamType() below.
```javascript
function testParamClass(param, types){
  for (var i = 0; i < types.length; i++) {
    if (param.name === types[i]) {
      return true;
    }
    else if (param.name === 'Constant') {
      return true;
    }
  }
  return false;
}
```